components/modal/README.md

```diff
@@ -0,0 +1,170 @@
+Modal
+=======
+
+The modal is used to create an accessible modal over an application.
+
+**Note:** The API for this modal has been mimicked to resemble `react-modal`.
+
+## Usage
+
+Render a screen overlay with a modal on top.
+```jsx
+	<Modal
+		title="My Modal"
+		onRequestClose={ closeFunction }
+		isOpen={ openState }
+		>
+		<ModalContent>
+			<p id="modal-description">This modal is meant to be awesome!</p>
+		</ModalConent>
+	</Modal>
+```
+
+## Implement close logic
+
+For the modal to properly work it's important you implement the close logic for the modal properly. The following example shows you how to properly implement a modal.
+
+```js
+const { Component, Fragment } = wp.element;
+const { Modal } = wp.components;
+
+class MyModalWrapper extends Component {
+	constructor() {
+		super( ...arguments );
+		this.state = {
+			isOpen: true,
+		}
+	}
+
+	closeModal() {
+		if ( this.state.isOpen ) {
+			this.setState( { isOpen: false } );
+		}
+	}
+	
+	openModal() {
+		if ( ! this.state.isOpen ) {
+			this.setState( { isOpen: true } );
+		}
+	}
+
+	render() {
+		return (
+			<Fragment>
+				<button onClick={ this.openModal.bind( this ) }>Open Modal</button>
```

Comment:
We're encouraging a bad (yet sometimes contested) behavior of creating new functions each `render` invocation.

See also: https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-no-bind.md